.github/workflows/include_guards.yml

```diff
@@ -0,0 +1,19 @@
+name: include guards
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - uses: sbeyer/include-guards-check-action@v2.0.0
+      with:
+        path: 'include/exadg/'
+        pattern: 'INCLUDE_EXADG_{dirs}_{file}_'
+        
```

Comment:
this defines the format of the include guard, which I deduced from the used definitions prior to this PR.
The `_` in the end was used in most cases, hence I considered it for the standard.